File: contest/aizu/ITP1_1_B/argc.cpp
#include <cstdio>

int main(int argc, char** argv) {
    int i;
    std::printf("argc: %d\n", argc);
    for (i = 0; i < argc; i++) {
        std::printf("argv[%d]: %s\n", i, argv[i]);
    }
    return 0;
}

File: src/haskell/argc.hs
#!/usr/bin/env runhaskell
module Main (main) where
import qualified System.Environment

main :: IO ()
main = do
  args <- System.Environment.getArgs
  putStrLn ("argc: " ++ (show . length) args)
  putStr (getString args 1)
  where
    getString :: [String] -> Int -> String
    getString [] _ = ""
    getString (x:xs) acc = "argv #" ++ show acc ++ ": " ++ x ++ "\n" ++ getString xs (acc + 1)
